rust/actyx/certs/src/bin/ax-cert.rs

```diff
@@ -32,13 +34,97 @@ struct AppLicenseOpts {
 
     /// ISO 8601 (i.e. 2014-11-28T12:00:09Z) expiration date time
     #[structopt(long)]
-    expires_at: DateTime<Utc>,
+    expires_at: Option<DateTime<Utc>>,
+
+    #[structopt(long, parse(try_from_str = parse_expires_in))]
+    expires_in: Option<DateTime<Utc>>,
 
     /// Requester's email address
     #[structopt(long)]
     email: String,
 }
 
+/// Parsing function for the `expires_in` variable.
+/// Accepts strings matching the regex: ^([0-9]+(Y|M|w|d|h|m|s))+$
+///
+/// Y - year(s) - counted as 365 days
+/// M - month(s) - counted as 30 days
+/// w - week(s)
+/// d - day(s)
+/// h - hour(s)
+/// m - minute(s)
+/// s - second(s)
+fn parse_expires_in(expires_in: &str) -> Result<DateTime<Utc>, anyhow::Error> {
+    parse_expires_in_as_duration(expires_in).map(|d| Utc::now() + d)
+}
+
+// This function is much easier to test for correctness
+fn parse_expires_in_as_duration(expires_in: &str) -> Result<chrono::Duration, anyhow::Error> {
+    lazy_static! {
+        static ref RE: Regex = Regex::new(r"\s*((?P<years>[0-9]+)Y)?\s*((?P<months>[0-9]+)M)?\s*((?P<weeks>[0-9]+)w)?\s*((?P<days>[0-9]+)d)?\s*((?P<hours>[0-9]+)h)?\s*((?P<minutes>[0-9]+)m)?\s*((?P<seconds>[0-9]+)s)?\s*").unwrap();
```

Comment:
perhaps split across multiple lines:
```suggestion
        static ref RE: Regex = Regex::new(r"\s*\
          ((?P<years>[0-9]+)Y)?\s*\
          ((?P<months>[0-9]+)M)?\s*\
          ((?P<weeks>[0-9]+)w)?\s*\
          ((?P<days>[0-9]+)d)?\s*\
          ((?P<hours>[0-9]+)h)?\s*\
          ((?P<minutes>[0-9]+)m)?\s*\
          ((?P<seconds>[0-9]+)s)?\s*").unwrap();
```

Reply:
Would've been amazing if it worked. With raw string literals it doesn't. But I found another solution